Reset Transport *before* Tests

rodnaph opened the pull request:
Updates the `InteractsWithMessenger` trait to clear the `TestTransport` **before** each test, as well as after.

Previously, the transport was only reset after each test in a TestCase *using this trait*. If other tests run first though which send messages, but *do not* make messenger assertions (as the transport is overridden for the entire `test` environment), then the transport isn't cleared at all, so the first test to run in a test case *with this trait* will have the state from the previous tests.

I don't know if there's currently a reason for having it specifically *after*, is this a BC break? Is there a better solution? Is there a test I can add or update to verify this behaviour?

rodnaph committed:
also reset transport before tests

Previously the transport was only reset after tests in a testcase using this
trait were run. This can cause an issue though if other tests
run first which send messages, but do not use this trait. In
this case the transport isn't cleared so the first test to run in a test
case *with* this trait will have the state from the previous tests.

kbond commented on src/InteractsWithMessenger.php:
```diff
@@ -13,7 +13,7 @@ trait InteractsWithMessenger
     /**
      * @internal
-     * @after
+     * @before
```
```suggestion
     * @after
     * @before
```

rodnaph replied:
👍

kbond reviewed (changes requested):
Resetting before makes sense for the reason you describe. We should keep the `@after` though.

BTW, I merged #39, so if you rebase - the tests should pass.

kbond commented:
I believe #24 would also solve this issue but let's go with this PR for now.

rodnaph commented:
> Resetting before makes sense for the reason you describe. We should keep the `@after` though.
> 
> BTW, I merged #39, so if you rebase - the tests should pass.

Updated and rebased, thanks.

kbond commented:
Thanks @rodnaph!
